src/main/scala/ai/privado/languageEngine/ruby/passes/PrivadoRubyTypeRecoveryPassGenerator.scala

```diff
@@ -131,17 +136,41 @@ private class RecoverForRubyFile(
     }
   }
 
+  def isCallParentScopeResolutionMatching(cName: String, code: String) = {
+    try {
+      val cNameList = cName.split(":program").last.split("\\.").filterNot(_.isEmpty)
+      val codeList  = code.split("\\(").head.split("[:.]").filterNot(_.isEmpty).dropRight(1).toList
+      cNameList sameElements codeList
+    } catch {
+      case e: Exception => false
+    }
+
+  }
+  def isCallHeadArgumentAScopeResolutionAndIsLastArgumentInTable(c: Call): Boolean = c.argument.headOption
+    .exists(_.isCall) && c.argument.head
+    .asInstanceOf[Call]
+    .name
+    .equals("<operator>.scopeResolution") && c.argument.head
+    .asInstanceOf[Call]
+    .argument
+    .lastOption
+    .exists(arg =>
+      symbolTable.get(arg).union(globalSymbolTable.get(arg)).exists(isCallParentScopeResolutionMatching(_, c.code))
```

Comment:
nit: maybe we add a comment in statements (or maybe functions) like these. Took me a while to figure this out 🤔